src/libLookup/Lookup.cpp

```diff
@@ -204,26 +204,35 @@ void Lookup::GetInitialBlocksAndShardingStructure() {
     }
     LOG_GENERAL(INFO,
                 "TxBlockNum " << txBlockNum << " DSBlockNum: " << dsBlockNum);
+    {
+      std::unique_lock<std::mutex> cv_lk(m_mutexCvSetRejoinRecovery);
+      m_rejoinRecoverySignal = false;
```

Comment:
Why set this here in a separate block and not right after line 211/215 below?

Reply:
Because `ComposeAndSendGetDirectoryBlocksFromSeed` call below might result in the response being sent straight away, and we better have the mutex unlocked.  But even if not, this is a function that may take some time, and we shouldn't be held the mutex aquired during its execution.